scripts/tasks/task-loadfw.py

```diff
@@ -0,0 +1,157 @@
+    
+    if verbose:
+        print(f"Exit code: {result.returncode}")
+        print(f"Output: {result.stdout}")
+        if result.returncode != 0:
+            print(f"Error: {result.stderr}")
+
+    result.check_returncode()
+
+
+def resolve_source_build(source_build):
+    source_build_dir = Path(source_build).parent
+    source_bins = [f for f in source_build_dir.glob('*.bin')]
+    try:
+        source_build_ts = os.path.getmtime(source_build)
+        newer_bins = [f for f in source_bins if os.path.getmtime(f) > source_build_ts]
+        if len(newer_bins) > 0:
+            raise RuntimeError("Found newer bin files")
+    except Exception as e:
+        def iter_bins():
+            bins = sorted(source_bins, key=lambda b: -os.path.getmtime(b)) 
+            for bin in bins:
+                print(bin)
+                yield "\t".join([str(datetime.datetime.fromtimestamp(os.path.getmtime(bin))), str(bin)])
+        chosen_build = iterfzf(iter_bins(), prompt=f"{e}, please confirm your build file.")
+        source_build = chosen_build.split("\t")[1]
+    return source_build
+
+def load_sd(args) -> None:
+    # only list fstype=msdos
+    device_name = args.name
+    if device_name is None:
+        disk_candidates = [part for part in psutil.disk_partitions() if part.fstype == 'msdos']
+        if len(disk_candidates) == 0:
+            raise RuntimeError("No deluge-compatible partitions found")
+        print(f"Available disks: {disk_candidates}")
+        def iter_disks():
+            for disk in disk_candidates:
+                yield disk.mountpoint
+
+        device_name = iterfzf(iter_disks())
+
+    source_build = resolve_source_build(args.build)
+    target_bins = [f for f in Path(device_name).glob("*.bin")]
+    target_build = 'deluge.bin'
+    try:
+        if len(target_bins) > 1:
+            raise RuntimeError("Found multiple target overwrite candidates")
+    except Exception as e:
+        def iter_target_bins():
+            for bin in target_bins:
+                yield bin
+        target_build = iterfzf(iter_target_bins(), prompt=f"{len(target_bins)} targets , please confirm your target build file.")
+
+
+    # check if there is a newer ".bin" file in the same directory as args.build 
+    # if so, confirm with the user if they are sure
+    # if not, show them a list of bin files and their modified dates via iterfzf
+    # set build to their choice
+
+    try:
+        dest = shutil.copy(source_build, os.path.join(device_name, target_build))
+        source_ts = str(datetime.datetime.fromtimestamp(os.path.getmtime(source_build)))
+        target_ts = str(datetime.datetime.fromtimestamp(os.path.getmtime(dest)))
+            
+        print(f"Copied {source_build} [{source_ts}] to {device_name} {dest} [{target_ts}]")
+    except:
+        print(f"Failed to copy {source_build} to {device_name}")
+        raise
+
+def load_sysex(args) -> None:
+    loadfw_path = Path("./toolchain/darwin-arm64/loadfw")
```

Comment:
we don't want to hardcode `darwin-arm64` as it might not exist, maybe a generic `toolchain/host` folder for things that are compiled on the host (or get the right toolchain name from some ENV var, I guess)

Reply:
THANK YOU! omg.... i'm so silly